clang/lib/CIR/Lowering/DirectToLLVM/LowerToLLVM.cpp

```diff
@@ -612,10 +612,16 @@ class CIRBrCondOpLowering
                   mlir::ConversionPatternRewriter &rewriter) const override {
     mlir::Value i1Condition;
 
+    auto hasOneUse = false;
+
+    if (auto defOp = brOp.getCond().getDefiningOp())
+      hasOneUse = defOp->getResult(0).hasOneUse();
+
     if (auto defOp = adaptor.getCond().getDefiningOp()) {
       if (auto zext = dyn_cast<mlir::LLVM::ZExtOp>(defOp)) {
         if (zext->use_empty() &&
-            zext->getOperand(0).getType() == rewriter.getI1Type()) {
+            zext->getOperand(0).getType() == rewriter.getI1Type() &&
+            hasOneUse) {
```

Comment:
Should the `hasOneUse` condition only guard the erase? We'd still want to use the `zext` operand as our `i1Condition` even if there's other uses, right?

Reply:
You're right! We still want to use the `i1condition` even if the `zext` has multiple uses. I have updated the file.